### ISSUE TITLE
Handling corrupted data

### ISSUE THREAD

**Issue by airknights** [pda/pheanstalk#111]
We had incidents of corrupted data, most probably due to network failure during job entry into queue.  It cut off the job string data about half way.  This resulted in getting the Pheanstalk_Exception_ClientException when CRLF wasn't the last character in the data.

This should be handled at the parser level, as it is still reserving the job, but we don't have a job id to handle this exception gracefully.  A parser should initiate the first line as "job meta data parser" and then do the data part as "job data parser".  Thereby providing the job id of a successful reservation for example.  This allows us to decide to do something else with the job, such as bury it or delete it, or report it in so we can resolve network issues.

Please note the issue I'm talking about is Connection::dispatchCommand function.

**Comment by SamMousa** [comment]
@airknights is this issue still relevant?
I'm thinking that if the data got corrupted when entering it into the queue, it is beanstalks responsibility.
If our connection to beanstalk fails when retrieving the data then that's another thing.

**Comment by gopi-ar** [comment]
We've seen this often and there's some correlation between unplanned system reboots / beanstalk crashes. We have to telnet into beanstalkd and delete the offending job. Not sure how other clients handle this error, but pheanstalk should probably deal with it gracefully.

**Comment by SamMousa** [comment]
Are you guys using persistent connections? If so, please turn try to reproduce without persistent connections enabled.

**Comment by gopi-ar** [comment]
On the servers we've seen this issue in, we've used pheanstalk in our long-running workers with persistent set to off.

**Comment by airknights** [comment]
This is old issue and is no longer relevant to me.

**Comment by SamMousa** [comment]
@gopi-ar when was the last time you saw this?
If it was a long time ago I'd prefer to close this issue for now.

**Comment by georaldc** [comment]
> We've seen this often and there's some correlation between this issue's occurrence and unplanned system reboots / beanstalk crashes. We have to telnet into beanstalkd and delete the offending job. Not sure how other clients handle this error, but pheanstalk should probably deal with it gracefully.

Sorry to comment on this closed issue, but I just encountered this problem last night. I am using an older v4 version of pheanstalk though due to current usage of PHP 7.3, but connecting to the latest version of beanstalkd. We've been inserting hundreds to thousands of jobs at any given moment for a few years now and a single job somehow got queued where the data was showing as truncated on beanstalkd. Interestingly, after I manually connected to the server and deleted the offending job (which was the only way I could fix this since pheanstalk complains about the missing CRLF bytes and other tools like beanstalk_console do not show the corrupted job at all and also complain similarly), about 3 minutes later beanstalkd crashed with a "segfault, error 4" error. I've seen this happen several times now, where it segfaults once every month or so, so I'm not sure if the corrupted job is somehow related to this (like something building up on beanstalkd that eventually leads to it having a segmentation fault after running nonstop for some time).

Since I use v4, no persistent connections are involved. I did not get any exceptions at the time the corrupted data was inserted so I don't think the SocketSocket implementation was made aware of any failures during its write loop calling socket_write()
